### ISSUE TITLE
AI Commander attacks from perceived closest base

### ISSUE THREAD

**Issue by MaszoPok** [Sparker95/Vindicta#298]
The AI Commander will attack from a base which it perceives as the closest base available. Ignoring terrain barriers like water or impassible terrain. leading to a longer route then a from another available base.

**Comment by billw2012** [comment]
Generate a low res nav grid to determine distance using pathfinding. Repurpose the existing virtualized pathfinder.

**Comment by Sparker95** [comment]
@billw2012 We just need to make very sure that it gives result extremely fast. It is being called per every action evaluation.

**Comment by billw2012** [comment]
Yeah i hope grid would be on the order of 20-50 nodes only, should be fast. We can even just make it a n x n grid lookup table actually for every grid to every other. We don't need the route only the distance. Map is 30x30 = 900 grid squares, 900 x 900 routes = 810000 element lookup table. We can cull all the ones that don't differ from straight line distance by more than x% which will be the vast majority of them.
